src/yival/enhancers/utils.py

```diff
@@ -28,22 +28,33 @@ def scratch_variations_from_str(
     target_str:
         This is the generated new output
         var1: hello world
+        hello world
         var2: bye
     
     variations: ["var1", "var2"]
 
     function result:
     {
-        "var1": "hello world"
+        "var1": "hello world\nhello world"
         "var2": "bye"
     }
     """
     result = {}
     lines = target_str.split('\n')
+    var_detect = False
+    var_now = ""
     for line in lines:
+        var_line = False
         for var in variations:
             if line.startswith(var + '='):
```

Comment:
improve readability and performance using f-strings: 
if line.startswith(f"{var}="):